tests/check/check_test.go

```diff
@@ -100,3 +106,197 @@ func testBadAuthModelID(t *testing.T, engine string) {
+}
+
+func BenchmarkCheckMySQL(b *testing.B) {
+	benchmarkAll(b, "mysql")
+}
+
+func benchmarkAll(b *testing.B, engine string) {
+	b.Run("BenchmarkCheckWithoutTrace", func(b *testing.B) { benchmarkCheckWithoutTrace(b, engine) })
+	b.Run("BenchmarkCheckWithTrace", func(b *testing.B) { benchmarkCheckWithTrace(b, engine) })
+	b.Run("BenchmarkCheckWithDirectResolution", func(b *testing.B) { benchmarkCheckWithDirectResolution(b, engine) })
+}
+
+const githubModel = `
+type user
+type team
+  relations
+    define member: [user,team#member] as self
+type repo
+  relations
+    define admin: [user,team#member] as self or repo_admin from owner
+    define maintainer: [user,team#member] as self or admin
+    define owner: [organization] as self
+    define reader: [user,team#member] as self or triager or repo_reader from owner
+    define triager: [user,team#member] as self or writer
+    define writer: [user,team#member] as self or maintainer or repo_writer from owner
+type organization
+  relations
+    define member: [user] as self or owner
+    define owner: [user] as self
+    define repo_admin: [user,organization#member] as self
+    define repo_reader: [user,organization#member] as self
+    define repo_writer: [user,organization#member] as self
+`
+
+func benchmarkCheckWithoutTrace(b *testing.B, engine string) {
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = engine
+
+	cancel := tests.StartServer(b, cfg)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(b, err)
+	defer conn.Close()
+	client := pb.NewOpenFGAServiceClient(conn)
+
+	ctx := context.Background()
+	resp, err := client.CreateStore(ctx, &pb.CreateStoreRequest{Name: "check benchmark without trace"})
+	require.NoError(b, err)
+
+	storeID := resp.GetId()
+	_, err = client.WriteAuthorizationModel(ctx, &pb.WriteAuthorizationModelRequest{
+		StoreId:         storeID,
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+		TypeDefinitions: parser.MustParse(githubModel),
+	})
+	require.NoError(b, err)
+	_, err = client.Write(ctx, &pb.WriteRequest{
+		StoreId: storeID,
+		Writes:  &pb.TupleKeys{TupleKeys: tuples},
+	})
+	require.NoError(b, err)
+
+	for i := 0; i < b.N; i++ {
+		_, err = client.Check(ctx, &pb.CheckRequest{
+			StoreId:  storeID,
+			TupleKey: tuple.NewTupleKey("repo:openfga/openfga", "reader", "user:github|iaco@openfga"),
+		})
+
+		require.NoError(b, err)
+	}
+}
+
+func benchmarkCheckWithTrace(b *testing.B, engine string) {
+	cfg := run.MustDefaultConfigWithRandomPorts()
+	cfg.Log.Level = "none"
+	cfg.Datastore.Engine = engine
+
+	cancel := tests.StartServer(b, cfg)
+	defer cancel()
+
+	conn, err := grpc.Dial(cfg.GRPC.Addr,
+		grpc.WithBlock(),
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	)
+	require.NoError(b, err)
+	defer conn.Close()
+	client := pb.NewOpenFGAServiceClient(conn)
```

Comment:
L194-L207 is all boilerplate code that is used in each `benchmarkCheck...` benchmark function. Can we consolidate please?

Reply:
done